src/node.cc

```diff
@@ -409,6 +409,10 @@ MaybeLocal<Value> StartExecution(Environment* env, StartExecutionCallback cb) {
     return StartExecution(env, "internal/main/watch_mode");
   }
 
+  if (!first_argv.empty() && first_argv == "run") {
```

Comment:
You did not introduce this syntax, so it is fine, but I am not sure why we do

```C++
!first_argv.empty() && first_argv == "run"
```

Instead of 

```C++
first_argv == "run"
```

It is not safety, for sure.

Maybe an attempt at optimization ?

I would be *very* surprised if it provided any kind of performance boost. Why would it? The logical way to compile this is to fix check the length of the string, see whether it is 3, and if so, do the string comparison. So why would you add an intermediate step where you check whether the string is empty? It makes no sense to me.

I recommend you do not change anything following my comment. You are following the existing style... but it is still weird and should maybe be revisited.

cc @joyeecheung

Reply:
I also don't know why it is `std::string first_argv` and not `std::string_view first_argv`. Not that it matters much since the string is always surely going to be allocated on stack (small string optimization), but if you are not going to mutate the string, why copy in the first place?

cc @joyeecheung

Reply:
I'll follow up on another pull-request

Reply:
I actually don't know why this isn't just `first_argv == "run"`, maybe it's just copy-pasta?

re. why it's not string_view: I think the code predates C++17 support.